### ISSUE TITLE
Add `Control.IndexOutOfBounds` effect

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#6009]
```
mod Control {
    pub eff IndexOutOfBounds {
        pub def new(index: Int32, min: Int32, max: Int32): Void
    }
}
```

And use it for partial functions:

- [ ] `Array.get`
- [ ] `Array.put`
- [ ] `Array.copyOfRange` (?)
- [ ] `Array.patch` (?)
- [ ] `Array.updateSequence` (?)

Same for `Vector`. What about `String`? What about other data types?

**Comment by JonathanStarup** [comment]
Master now has `OutOfBounds`, but it hasn't been fully integrated other than `Indexable`

**Comment by magnus-madsen** [comment]
Yeah, eventually it should arrive for some Array functions I think

**Comment by magnus-madsen** [comment]
Fixed in master.
